frontend/src/app/main/ui/auth/register.cljs

```diff
@@ -65,23 +114,60 @@
 
                (st/emit! (ntf/error (tr "errors.generic")))))))
 
+        on-success
+        (mf/use-fn
+         (mf/deps on-success-callback)
+         (fn [params]
+           (if (fn? on-success-callback)
+             (on-success-callback (:email params))
+
+             (cond
+               (some? (:token params))
+               (let [token (:token params)]
+                 (st/emit! (rt/nav :auth-verify-token {:token token})))
+
+               (:is-active params)
+               (st/emit! (da/login-from-register))
+
+               :else
+               (do
+                 (swap! storage/user assoc ::email (:email params))
+                 (st/emit! (rt/nav :auth-register-success)))))))
+
+        on-register-profile
+        (mf/use-fn
+         (mf/deps on-success on-error)
+         (fn [form]
+           (reset! submitted? true)
+           (let [create-welcome-file?
+                 (cf/external-feature-flag "onboarding-03" "test")
+
+                 params
+                 (cond-> form
+                   create-welcome-file? (assoc :create-welcome-file true))]
+             (->> (rp/cmd! :register-profile params)
+                  (rx/finalize #(reset! submitted? false))
+                  (rx/subs! on-success on-error)))))
```

Comment:
I think you can pass the finalize callback as third argument to rx/subs! instead of rx/finalize here